omod-common/src/test/java/org/openmrs/module/webservices/rest/web/api/impl/RestServiceImplTest.java

```diff
@@ -147,6 +166,178 @@ public void getRepresentation_shouldReturnAnInstanceOfNamedRepresentationForGive
 	}
 	
 	/**
+	 * @verifies return resource for given name
+	 * @see RestServiceImpl#getResourceByName(String)
+	 */
+	@Test
+	public void getResourceByName_shouldReturnResourceForGivenName() throws Exception {
+		
+		List<Class<? extends Resource>> resources = new ArrayList<Class<? extends Resource>>();
+		resources.add(MockingBirdFantasyResource.class);
+		
+		when(openmrsClassScanner.getClasses(Resource.class, true)).thenReturn(resources);
+		setCurrentOpenmrsVersion(EXISTING_RESOURCE_SUPPORTED_OPENMRS_VERSION);
+		
+		assertThat(restService.getResourceByName("v1/mockingbirdfantasy"), instanceOf(MockingBirdFantasyResource.class));
+	}
+	
+	/**
+	 * @verifies return resource for given name and ignore unannotated resources
+	 * @see RestServiceImpl#getResourceByName(String)
+	 */
+	@Test
+	public void getResourceByName_shouldReturnResourceForGivenNameAndIgnoreUnannotatedResources() throws Exception {
+		
+		List<Class<? extends Resource>> resources = new ArrayList<Class<? extends Resource>>();
+		resources.add(MockingBirdFantasyResource.class);
+		resources.add(UnannotatedMockingBirdFantasyResource.class);
+		
+		when(openmrsClassScanner.getClasses(Resource.class, true)).thenReturn(resources);
+		setCurrentOpenmrsVersion(EXISTING_RESOURCE_SUPPORTED_OPENMRS_VERSION);
+		
+		assertThat(restService.getResourceByName("v1/mockingbirdfantasy"), instanceOf(MockingBirdFantasyResource.class));
+	}
+	
+	/**
+	 * Helper method to set the current OpenMRS version for tests.
+	 * 
+	 * @param currentOpenmrsVersion the openmrs version to set the current version to
+	 * @throws NoSuchFieldException
+	 * @throws IllegalAccessException
+	 */
+	private void setCurrentOpenmrsVersion(final String currentOpenmrsVersion) throws NoSuchFieldException,
+	        IllegalAccessException {
+		
+		Field versionField = OpenmrsConstants.class.getDeclaredField("OPENMRS_VERSION_SHORT");
+		Field modifiersField = Field.class.getDeclaredField("modifiers");
+		modifiersField.setAccessible(true);
+		modifiersField.setInt(versionField, versionField.getModifiers() & ~Modifier.FINAL);
+		versionField.set(null, currentOpenmrsVersion);
+	}
+	
+	/**
+	 * @verifies fail if failed to get resource classes
+	 * @see RestServiceImpl#getResourceByName(String)
+	 */
+	@Test
+	public void getResourceByName_shouldFailIfFailedToGetResourceClasses() throws Exception {
+		
+		IOException ioException = new IOException("some");
+		
+		when(openmrsClassScanner.getClasses(Resource.class, true)).thenThrow(ioException);
```

Comment:
For the above, why not an empty list? That way we are sure that an exception is thrown by the code for a non existing resource.